mwclient/client.py

```
@@ -969,7 +969,21 @@ def ask(self, query, title=None):
         API doc: https://semantic-mediawiki.org/wiki/Ask_API
 
         Returns:
-            Generator for retrieving all search results
+            Generator for retrieving all search results.
+            Iterating over the generator gives a dictionnary containing the answers
+            to the query.
+            The dictionnary is empty if either:
+               - the query is valid but there are no answers,
+               - or if the query is not valid.
```

This should probably raise an exception, see below